### DISCUSSION TITLE
Request: more theme customization options

### DISCUSSION THREAD

**Discussion by SteveOnorato** [Able-Bear-Software/tab-nodes-tree#18] [Ideas / Feature Request]
1. Can we have separate styles (tab background & text settings) for unread tabs?  Same for unloaded tabs.
2. In addition to the text color choices, I'd also like the ability to change the text to bold and/or italic.

Thanks!

**Comment by SteveOnorato** [comment]
For reference, here are some of the CSS customizations that make the Firefox "Tree Style Tab" extension great for me:
```
/* Show title of unread tabs with red and italic font */
.tab.unread .label {
  color: red !important;
  font-style: italic !important;
}

/* Only show tab close button on active tab
https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#only-show-tab-close-button-on-hover-1448 */
/* "#tabbar" is required for TST 3.4.0 and later! */
#tabbar tab-item:not(.active) tab-closebox {
  display: none;
}

/* Make the active tab yellow with bold text */
#tabbar tab-item.active tab-item-substance {
  background-color: rgb(255,255,195);
  font-weight: bold;
}

/* Fade the background for unloaded tabs and show the title in italics */
.tab.discarded {
  opacity: 0.93;
}
.tab.discarded tab-item-substance {
  background-color: rgb(208,208,208);
  font-style: italic !important;
}
```

**Comment by zeroarst** [comment]
Thanks for your feedback! Allowing users to customize styles for different tab statuses is on my to-do list.
https://github.com/Able-Bear-Software/tab-nodes-tree/issues/15
https://github.com/Able-Bear-Software/tab-nodes-tree/issues/19

**Comment by zeroarst** [comment]
@SteveOnorato I just created a quick prototype, and I’d love to get your feedback. Please note that this is not the final version.

In this design, both the Dark and Light themes will have their own tab node content style settings. Additionally, for customized themes, users can choose to use the existing dark/light themes or configure their own.

Unread and discard has own styles.

![image](https://github.com/user-attachments/assets/0662a202-ec70-478a-bcdd-59319f9493fb)

**Reply by zeroarst** [reply]
Wondring if changing background color is necessary. For it seems opacity is enough.

**Reply by SteveOnorato** [reply]
I don't use dark mode, but this seems reasonable.

Hopefully the text styles are additive; a discarded and unread tab would have its text in both bold and italics in the example above.
But what about the color?  Should the user be able to reorder these so they can pick which one has priority?

**Comment by zeroarst** [comment]
@SteveOnorato 

95% progress… doing more testing.

There will be two new colors for each theme—`unread` and `discarded`—added to both `dark` and `light` themes as well as customized themes.

For the `discarded` and `unread` styles, `discarded` will take precedence over `unread`.

Customized themes can opt to override the dark/light theme tab node `unread`/`discarded` styles. This is particularly useful for users with multiple themes who don’t want to set different statuses for each. They can disable the customized theme tab node `unread`/`discarded` styles to stick with the dark/light theme ones.

**Summary:**
- **Icon and text color priority:** `Discarded` > `Unread` > `Highlighted` > `Selected` > `Active` > `Normal`
- **Tab node styles priority:** Customized theme `Discarded` (if enabled) > Dark/Light theme `Discarded` > Customized theme `Unread` (if enabled) > Dark/Light theme `Unread`.

![image](https://github.com/user-attachments/assets/aceaaf27-af71-4907-ae7d-78a287d0a1ad)
![image](https://github.com/user-attachments/assets/4ae09733-b811-453e-bfd8-3c37b9880781)

**Comment by zeroarst** [comment]
Release in 1.4.0.
- #19 
- #15
